tests/test_pre_commit.py

```diff
@@ -364,3 +372,171 @@ def test_wrap_markdown_cell(tmpdir):
     text = nb.cells[0].source
     assert len(text.splitlines()) >= 2
     assert text != long_text
+
+
+def test_is_untracked(tmpdir):
+    git = git_in_tmpdir(tmpdir)
+
+    # make a test file
+    file = tmpdir.join("test.txt")
+    file.write("test file\n")
+    file = str(file)
+
+    with tmpdir.as_cwd():
+        # untracked
+        assert is_untracked(file)
+
+        # added, not committed
+        git("add", file)
+        assert not is_untracked(file)
+
+        # committed
+        git("commit", "-m", "'test'")
+        assert not is_untracked(file)
+
+
+def test_add_untracked_adds_new(tmpdir):
+    git = git_in_tmpdir(tmpdir)
```

Comment:
with so many calls to git like this, why not just gitpython or some other nice bindings. :)

Reply:
Agreed, I was just using the style that was already there, I'm not here to refactor the test suite

Reply:
Ah, note to @mwouts then.

Reply:
Oh yes I'm sorry I'm just discovering `gitpython`... I'll refactor that part later on :smile: